packages/core/src/render3/component_ref.ts

```diff
@@ -163,6 +164,17 @@ export class ComponentFactory<T> extends AbstractComponentFactory<T> {
       injector: Injector, projectableNodes?: any[][]|undefined, rootSelectorOrNode?: any,
       environmentInjector?: NgModuleRef<any>|EnvironmentInjector|
       undefined): AbstractComponentRef<T> {
+    // Check if the component is orphan
+    if (ngDevMode && this.componentDef.debugInfo?.forbidOrphanRendering) {
+      if (depsTracker.isOrphanComponent(this.componentType)) {
+        throw new RuntimeError(
+            RuntimeErrorCode.RUNTIME_DEPS_ORPHAN_COMPONENT,
+            `Orphan component found! Trying to render the component ${
+                debugStringifyTypeForError(
+                    this.componentType)} without first loading the NgModule that declares it. It is recommended to make this component standalone in order to avoid such confusing cases. If this is not possible now, please import the component's NgModule in the NgModule or the standalone component in which you are trying to render this component. Also make sure the way the app is bundled and served always includes the component's NgModule before the component.`);
```

Comment:
```suggestion
                    this.componentType)} without first loading the NgModule that declares it. It is recommended to make this component standalone in order to avoid this error. If this is not possible now, import the component's NgModule in the appropriate NgModule, or the standalone component in which you are trying to render this component. If this is a lazy import, load the NgModule lazily as well and use its module injector.`);
```

I think it's important to call out the solution for lazy NgModule cases (which are not super uncommon). I'd drop the order of bundling. Feels a little too specific and most Angular users don't control bundling